app/src/main/java/org/gdg/frisbee/android/gde/GdeActivity.java

```diff
@@ -94,102 +84,66 @@ public void networkFailure(Throwable error) {
         });
     }
 
-    private void addGdes(final GdeList directory) {
-        mHandler.post(new Runnable() {
-            @Override
-            public void run() {
-                HashMap<String, GdeList> gdeMap = new HashMap<>();
-
-                for (Gde gde : directory) {
-                    if (gde.getProduct() == null) {
-                        continue;
-                    }
-                    for (String p : gde.getProduct()) {
-                        String product = p.trim();
-                        if (!gdeMap.containsKey(product)) {
-                            gdeMap.put(product, new GdeList());
-                        }
-
-                        gdeMap.get(product).add(gde);
-                    }
-                }
-
-                mViewPagerAdapter.addMap(gdeMap);
-                mViewPagerAdapter.notifyDataSetChanged();
-
-                mViewPager.setAdapter(mViewPagerAdapter);
-                mTabLayout.setupWithViewPager(mViewPager);
-
-                getAchievementActionHandler().handleLookingForExperts();
-            }
-        });
-    }
-
-    protected String getTrackedViewName() {
-        return null;
+    private void setupGdeViewPager(GdeList directory) {
```

Comment:
I've changed `addGdes` to `setupGdeViewPager`. Because here I do all the setup now. This way we are creating the adapter with a final immutable list.